components/block_v2/templates/block_no_kind.jade

```diff
@@ -0,0 +1,62 @@
+- visibilityClass = block.__typename === 'Channel' ? "Block__inner--channel--" + block.visibility : ''
+
+a.Block(
+  href=block.href
+  data-client=block.klass
+  data-id=block.id
+)
+  .Block__inner(
+    class="Block__inner--#{block.__typename.toLowerCase()} #{visibilityClass}"
+  )
+    //- Connect interface
+    .Block__inner__connect.js-connect-container
+      //- Rendered client-side
+
+    //- Hover overlay
+    .Block__inner__overlay
+      if block.__typename !== 'Channel'
+        if block.source_url || block.file_url
+          button.Button.Button--overlay.js-source Source
+        if user && user.has('username')
+          button.Button.Button--overlay.js-connect Connect →
+      else if user && user.has('username')
+        .Block__inner__overlay__channel-actions
+          button.Button.Button--divider.js-follow(class="Button--#{block.visibility}") Follow
+          button.Button.Button--divider.js-connect(class="Button--#{block.visibility}") Connect →
+
+    //- Main content
+    if block.__typename === 'Channel'
+      h2!= block.title
+      .Block__inner--channel__stats
+         ul.Block__inner--channel__stats__list
+          li by #{block.user.name}
+          li #{block.counts.blocks} blocks &bull; #{block.updated_at}
+
+    //- Text content
+    if block.__typename === 'Text'
+      .Block__inner--text__content
+        p!= block.content
+
+    //- Image / Attachment content
+    if block.file_extension && block.file_extension !== 'pdf'
+      img(
+        data-src="#{sd.IMAGE_PATH}iconic/file.svg"
+        data-file-extension=block.file_extension
+        class="iconic iconic-lg"
+      )
+    else if block.image_url
+      img(src=block.image_url)
+
+  //- Bottom caption
+  if block.__typename !== 'Channel'
+    .Block__caption
+      .Block__caption__title
+        = block.title
+        - if(block.__typename === 'Attachment' && ['mp3', 'pdf'].indexOf(block.file_extension))
+          img(data-src="#{sd.IMAGE_PATH}iconic/type-#{block.file_extension}.svg" class="iconic iconic-sm")
+        - if(block.__typename === 'Embed')
+          img(data-src="#{sd.IMAGE_PATH}iconic/type-embed.svg" class="iconic iconic-sm")
+
+      .Block__caption__meta
+        .Block__caption__meta__author Added by #{block.user.name}
+        .Block__caption__meta__timestamp Updated #{block.updated_at}
```

Comment:
Let us hope this page doesn't require significant maintenance! We should start a hitcounter for competing block implementations.